### PR TITLE
Config

### PR DESCRIPTION
**Pull request by ReihaneENS**
add : config

### CODE REVIEW

File: .env
Comment (feritka79) [inline .env:1]: Add this file to the gitignore

File: src/config/config.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import mongoose from "mongoose";
```
Comment (feritka79) [inline src/config/config.ts:1]: Change the file name to connectDb.
Use connect ( import { connect } from 'mongoose' )

File: src/config/config.ts [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+const connectDb = async () => {
+  try {
+    // await mongoose.connect(process.env.DB_URI as string);
```
Comment (feritka79) [inline src/config/config.ts:5]: use connect ( await connect(url, { dbName }) )
Take uri and dbName as parameters in the function

File: src/env.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import enforceEnv from 'envil';
+
+const checkEnvs = ()=>{
```
Comment (feritka79) [inline src/env.ts:3]: Change the function name to getEnvs

File: src/env.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    const envs=[
+        'DB_URI',
+        'DB_NAME',
+        'PORT',
+        'JWT_SECRET',
+        'EMAIL_HOST',
+        'EMAIL_PORT',
+    ];
+    enforceEnv(envs)
```
Comment (feritka79) [inline src/env.ts:12]: Use enforceEnv like this:
enforceEnv(envVariables, { returnValues: true });
and return it

File: src/index.ts [hunk 5]
```diff
@@ -1,12 +1,17 @@
-import express from 'express';
+import express from "express";
+import connectDB from "./config/config";
```
Comment (feritka79) [inline src/index.ts:2]: Add a blank line here

File: src/index.ts [hunk 6]
```diff
@@ -1,12 +1,17 @@
...
+import userRoutes from "./routes/userRoutes";
+import dotenv from "dotenv";
+
+dotenv.config();
```
Comment (feritka79) [inline src/index.ts:6]: remove dotenv.
When using getEnvs, there is no need for dotenv

File: src/index.ts [hunk 7]
```diff
@@ -1,12 +1,17 @@
...
 
 const app = express();
-const port = 3000;
 
-app.get('/a', (req, res) => {
-  res.send('???');
-});
+connectDB();
```
Comment (feritka79) [inline src/index.ts:10]: Pass these parameters like this :
const { DB_URI, DB_NAME } = getEnvs();
connectDb(DB_URI,DB_NAME)

File: src/index.ts [hunk 8]
```diff
@@ -1,12 +1,17 @@
...
+
+app.use(express.json());
+app.use("/users", userRoutes);
```
Comment (feritka79) [inline src/index.ts:13]: Create a routes folder and add routes there

File: src/index.ts [hunk 9]
```diff
@@ -1,12 +1,17 @@
...
+
+const PORT = process.env.PORT || 5000;
```
Comment (feritka79) [inline src/index.ts:15]: use PORT

const {PORT} = getEnvs()

File: tsconfig.json [hunk 10]
```diff
@@ -23,14 +23,14 @@
     // "moduleDetection": "auto",                        /* Control what method is used to detect module-format JS files. */
     /* Modules */
     "module": "commonjs" /* Specify what module code is generated. */,
-    // "rootDir": "./",                                  /* Specify the root folder within your source files. */
+    "rootDir": "./src",                                /* Specify the root folder within your source files. */
```
Comment (feritka79) [inline tsconfig.json:26]: No need for this property

File: tsconfig.json [hunk 11]
```diff
@@ -101,6 +101,8 @@
     // "skipDefaultLibCheck": true,                      /* Skip type checking .d.ts files that are included with TypeScript. */
     "skipLibCheck": true /* Skip type checking all .d.ts files. */
   },
-  "include": ["**/*"],
-  "exclude": ["dist"]
+  "include": [
+    "src/**/*.ts"
+  ],
+  "exclude": ["node_modules", "**/*.test.ts"]
```
Comment (feritka79) [inline tsconfig.json:107]: No need for this property
add property ( "outDir": "./dist")

**Review by feritka79 — CHANGES_REQUESTED** [review]

File: src/index.ts (hunk 8 above)
Comment (ceydae) [inline src/index.ts:13]: remove routes for now

**Review by ceydae — CHANGES_REQUESTED** [review]
Learn about the commit conventions in Notion.

Remove the `route(users)` because in this specific PR, we do not want to handle any routes and our only focus is the base config and files. 

There's also an error when trying to run the project, you should run `npm run build` before any PRs to check if there's anything wrong.

Currently, there's this error:

```
src/index.ts:3:24 - error TS2307: Cannot find module './routes/userRoutes' or its corresponding type declarations.

3 import userRoutes from "./routes/userRoutes";
```

File: src/index.ts [hunk 12]
```diff
@@ -1,12 +1,27 @@
...
+
+        const port = PORT || 5000;
+        app.listen(port, () => {
+            console.log(`Server is running on port ${port}`);
+        });
+    })
+    .catch(err => {
+        console.error('Failed to connect to the database:', err);
+    });
+
```
Comment (feritka79) [inline src/index.ts:25]: Modify the above code as follows:
connectDB(DB_URI,  DB_NAME)
app.listen(port, () => {
      console.log(`Server is running on port ${port}`)
});

**Avoid using route in this file**

File: src/index.ts [hunk 13]
```diff
@@ -1,12 +1,27 @@
...
+import connectDb from "./config/connectDb";
+
 
 const app = express();
-const port = 3000;
 
-app.get('/a', (req, res) => {
-  res.send('???');
-});
+const { DB_URI, DB_NAME } = getEnvs();
```
Comment (feritka79) [inline src/index.ts:9]: Retrieve the PORT here  { DB_URI, DB_NAME , PORT}

File: src/index.ts [hunk 14]
```diff
@@ -1,12 +1,27 @@
...
+    })
+    .catch(err => {
+        console.error('Failed to connect to the database:', err);
+    });
+
 
-app.listen(port, () => {
-  console.log(`Server running on port ${port}`);
-});
+const {PORT} = getEnvs();
```
Comment (feritka79) [inline src/index.ts:27]: remove this line

File: src/env.ts [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+    const envVariables = [
+        'DB_URI',
+        'DB_NAME',
+        'PORT',
+        'JWT_SECRET',
+        'EMAIL_HOST',
+        'EMAIL_PORT',
+    ];
+    const envValues = enforceEnv(envVariables, {returnValues: true});
+    return envValues;
```
Comment (feritka79) [inline src/env.ts:13]: Return the function directly.
No need to create a variable

**Review by feritka79 — CHANGES_REQUESTED** [review]

File: .env
Comment (ceydae) [inline .env:1]: ignore this file from git

File: src/env.ts
Comment (ceydae) [inline src/env.ts:1]: rename to src/getEnvs.ts

File: src/index.ts [hunk 16]
```diff
@@ -1,12 +1,21 @@
...
 
-app.listen(port, () => {
-  console.log(`Server running on port ${port}`);
-});
+        app.listen(PORT, () => {
+            console.log(`Server is running on port ${PORT}`);
+        });
+    })
+    .catch(err => {
+        console.error('Failed to connect to the database:', err);
```
Comment (ceydae) [inline src/index.ts:20]: No need to wait for DB to connect and then implement the routes. They can be written separately and asynchronously

Comment (ceydae) [reply src/index.ts:20]: Like so:

```
connectDB()

app.get()

app.listen()

**Review by ceydae — CHANGES_REQUESTED** [review]

**Comment by ceydae** [comment]
No need for connectDB() to have a `.then()` or `.catch()` because they are already handled in the connectDB function.

Also, resolve the conflicts.

**Pull request merged**